test/unit/task2/service/userService.test.js

```diff
@@ -0,0 +1,210 @@
+const userService = require('./../../../../src/task2/service/userService')
+const userRepository = require('./../../../../src/task2/repository/usersRepository')
+
+jest.mock('./../../../../src/task2/repository/usersRepository')
+
+const USER_ALREADY_EXISTS_MESSAGE = 'User with such id already exists'
+const USER_NOT_FOUND_MESSAGE = 'User not found'
+const USER_ID = 'id'
+const USER_LOGIN = 'login'
+const USER_PASSWORD = 'password'
+const USER_AGE = 'age'
+const IS_DELETED = false
+const payload = {
+    id: USER_ID,
+    login: USER_LOGIN,
+    password: USER_PASSWORD,
+    age: USER_AGE,
+    isDeleted: IS_DELETED
+}
+
+const user = {
+    id: USER_ID,
+    login: USER_LOGIN,
+    password: USER_PASSWORD,
+    age: USER_AGE,
+    isDeleted: IS_DELETED
+}
+
+afterAll(() => {
+    jest.resetAllMocks()
+})
+
+afterEach(() => {
+    jest.clearAllMocks()
+})
+
+describe('Create user', () => {
+    test('should return newly created user', async () => {
+        userRepository.getUser.mockResolvedValue(null)
+        userRepository.saveUser.mockResolvedValue(user)
+
+        const result = await userService.createUser(payload)
+
+        expect(result).toEqual(user)
+    })
+
+    test('should throw expected exception when user already exists', async () => {
+        const error = new Error(USER_ALREADY_EXISTS_MESSAGE)
+        userRepository.getUser.mockResolvedValue(user)
+
+        await expect(async () => {
```

Comment:
better to use try/catch 


```
try {
await userService.createUser(payload)
} catch (error) {
expect(error).toStrictEqual(expectedError)
}
```

use it to assert errors across all test files

Reply:
Fixed